internal/operators/ocs/validations.go

```diff
@@ -152,23 +158,29 @@ func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
 	err = setOperatorStatus(cluster, status)
 	if err != nil {
 		o.log.Error("Failed to set Operator status ", err)
-		return false
+		return "failure"
 	}
 
-	return canDeployOCS
+	if canDeployOCS {
+		return "success"
+	}
+	return "failure"
 }
 
-func (o *ocsValidator) nodeResources(host *models.Host, cpuCount *int64, totalRAM *int64, diskCount *int64, hostsWithDisks *int64, insufficientHosts *[]string) error {
+func (o *ocsValidator) nodeResources(host *models.Host, cpuCount *int64, totalRAM *int64, diskCount *int64, hostsWithDisks *int64, insufficientHosts *[]string) (bool, error) {
 	var inventory models.Inventory
-	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+	// if inventory is empty
+	if host.Inventory == "" {
+		return true, nil // to indicate that inventory is empty and the ValidationStatus must be Pending
```

Comment:
please add log that host inventory is empty with the host id details

Reply:
fixed